### PR TITLE
Bump kube-vip

### PR DESCRIPTION
**Pull request by yaocw2020**
Solution:
- The kube-vip could use the network interface based on the default route automatically. If the default routes changes, it will be panic and restart as a daemonset pod. Thus, we don't need a VIP controller in the harvester network controller to change the VIP interface of the kube-vip anymore. The advance is architecture decoupling to solve the problems like [1681](https://github.com/harvester/harvester/issues/1681)
- The kube-vip v0.4.1 supports static VIP configured in the field `.spec.loadBalancerIP` of the service.

Related issue: 
https://github.com/harvester/harvester/issues/1772
https://github.com/harvester/harvester/issues/1782
https://github.com/harvester/harvester/issues/1681

- Bump kube-vip v0.4.1 
- Change the way to configure static VIP

### CODE REVIEW

**Comment by gitlawr** [comment]
How does upgrade work with this change?

**Comment by yaocw2020** [comment]
> How does upgrade work with this change?

After discussion with @bk201, the upgrade component should add something to support the dependency chart parameters update in the harvester managedChart.

The related [issue](https://github.com/harvester/harvester/issues/1810) about upgrading.

File: deploy/charts/harvester/values.yaml [hunk 1]
```diff
@@ -429,9 +429,9 @@ kube-vip:
   enabled: false
   image:
     repository: plndr/kube-vip
-    tag: v0.3.7
-  config:
-    vip_interface: eth0
+    tag: main
+  nodeSelector:
+    node-role.kubernetes.io/control-plane: "true"
```
Comment (guangbochen) [inline deploy/charts/harvester/values.yaml:434]: why should we only deploy the `kubevip` pod to the control-plane node only?

Comment (bk201) [reply deploy/charts/harvester/values.yaml:434]: Just want to add we [suggested users join nodes with VIP for high availability](https://github.com/harvester/harvester/issues/1521).
HA in RKE2 doesn't work if VIP is on an agent node.

**Review by guangbochen — APPROVED** [review]
LGTM, and will need QA to validate the upgrade issue tracked on the  https://github.com/harvester/harvester/issues/1810

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**